### PR TITLE
initial draft of ADR030

### PR DESCRIPTION
**Pull request by pauldougan**
New ADR to document the rationale for running a single staging environment in London

What
----

A new Architecture Decision Record to document the decision to run a single staging environment in the London region.

How to review
-------------
```
bundle exec middleman build
bundle exec middleman server
```
Review content in the context of the team manual website for accuracy and completeness

Who can review
--------------

- Alex
- Henry

### CODE REVIEW

File: source/architecture_decision_records/ADR030-single-staging-london.html.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+# Status
+Accepted
+
+# Consequences
+- Fewer moving parts and reduced infrastructure footprint.
+- The change moves closer to our target of running our platform in London
+- Reduce the cost of operation.
+- Fewer environments and simplification of our pipelines.
+- We will check that we are able to run our pipelines to deploy to Ireland and London.
```
Comment (samcrang) [inline source/architecture_decision_records/ADR030-single-staging-london.html.md]: I don't understand what this is saying—is it meant to say "we will be unable to check that our pipelines deploy to both Ireland and London"?

File: source/architecture_decision_records/ADR030-single-staging-london.html.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
+# ADR030: Single staging environment in London
+
+
+# Context
+
+We run a Cloud Foundry platform in London (AWS eu-west-2 region) to satisfy GOV.UK PaaS tenant needs to host on UK soil. This is driven by regulatory compliance and the hosting needs of selected existing GOV.UK PaaS tenants who are making preparations for Brexit.
+
+Longer term we will migrate tenants to the London region and demise the region in Ireland to further reduce our footprint.
```
Comment (tlwr) [inline source/architecture_decision_records/ADR030-single-staging-london.html.md]: I am not sure you can use `demise` as a verb here. In this context AFAIK it means to move or transfer.

I think `turn off` or `destroy` is probably sufficient.

File: source/architecture_decision_records/ADR030-single-staging-london.html.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+# Status
+Accepted
+
+# Consequences
+- Fewer moving parts and reduced infrastructure footprint.
+- The change moves closer to our target of running our platform in London
+- Reduce the cost of operation.
+- Fewer environments and simplification of our pipelines.
+- We will check that we are able to run our pipelines to deploy to Ireland and London.
+- We will check the impact on our infrastructure usage.
```
Comment (samcrang) [inline source/architecture_decision_records/ADR030-single-staging-london.html.md]: I don't understand what this means and it seems really vague.

File: source/architecture_decision_records/ADR030-single-staging-london.html.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+
+We will demise the Ireland (eu-west-1) staging environment resulting in a single staging environment in AWS London.
+
+
+# Status
+Accepted
+
+# Consequences
+- Fewer moving parts and reduced infrastructure footprint.
```
Comment (samcrang) [inline source/architecture_decision_records/ADR030-single-staging-london.html.md]: I think I'd prefer not to use words like "footprint" as it seems a bit vague/business-y.

File: source/architecture_decision_records/ADR030-single-staging-london.html.md (hunk 2 above)
Comment (tlwr) [reply source/architecture_decision_records/ADR030-single-staging-london.html.md]: Also `we will migrate tenants` conflicts with our support model - AFAIK we (read: the PaaS team) will not be doing migration work on behalf of others.

File: source/architecture_decision_records/ADR030-single-staging-london.html.md [hunk 5]
```diff
@@ -0,0 +1,36 @@
+# ADR030: Single staging environment in London
+
+
+# Context
+
+We run a Cloud Foundry platform in London (AWS eu-west-2 region) to satisfy GOV.UK PaaS tenant needs to host on UK soil. This is driven by regulatory compliance and the hosting needs of selected existing GOV.UK PaaS tenants who are making preparations for Brexit.
```
Comment (samcrang) [inline source/architecture_decision_records/ADR030-single-staging-london.html.md]: Can we just say "We run a production Cloud Foundry deployment in London to allow tenants to host applications in the UK"?

File: source/architecture_decision_records/ADR030-single-staging-london.html.md (hunk 3 above)
Comment (tlwr) [inline source/architecture_decision_records/ADR030-single-staging-london.html.md]: I feel like these last two points are unnecessary

File: source/architecture_decision_records/ADR030-single-staging-london.html.md (hunk 5 above)
Comment (samcrang) [inline source/architecture_decision_records/ADR030-single-staging-london.html.md]: Is it possible to link out to the relevant regulatory compliance? Does it exist yet?

File: source/architecture_decision_records/ADR030-single-staging-london.html.md (hunk 2 above)
Comment (alext) [reply source/architecture_decision_records/ADR030-single-staging-london.html.md]: I agree that this ADR shouldn't cover any future plans relating to production - that's a separate concern. I think we should keep this focussed on the staging environments.

File: source/architecture_decision_records/ADR030-single-staging-london.html.md [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+Longer term we will migrate tenants to the London region and demise the region in Ireland to further reduce our footprint.
+
+In addition we need to responsibly manage our use of Amazon Web Service resources to reduce the ongoing running costs. Having duplicate staging environments doubles the amount of infrastructure and hence cost.
+
+The London region is newer than the Ireland and offers a slightly reduced subset of services compared to the more mature Ireland region.
```
Comment (tlwr) [inline source/architecture_decision_records/ADR030-single-staging-london.html.md]: I think `slightly` is unnecessary

Also `reduced subset` is redundant, I would go with `subset`, i.e. `The London region offers a subset of services offered in the Ireland region.`. (And maybe explain why this is important?)

With regards to referring to regions as more/less mature, this is confusing and AWS does not do this, we should not either. As an example: Oregon is newer than N. Virginia, but had EKS before, whereas N. Virginia had Fargate before Oregon. The age of the region is no indication of maturity/stability (see us-east-1 outages).

File: source/architecture_decision_records/ADR030-single-staging-london.html.md [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+
+We feel the risks of running a single staging environment are offset by the cost savings and the reduction in footprint.
+
+
+
+# Decision
+
+
+We will demise the Ireland (eu-west-1) staging environment resulting in a single staging environment in AWS London.
```
Comment (tlwr) [inline source/architecture_decision_records/ADR030-single-staging-london.html.md]: See above comments WRT to demise (however my English skills are _awful_)

File: source/architecture_decision_records/ADR030-single-staging-london.html.md [hunk 8]
```diff
@@ -0,0 +1,36 @@
+# ADR030: Single staging environment in London
+
+
+# Context
+
+We run a Cloud Foundry platform in London (AWS eu-west-2 region) to satisfy GOV.UK PaaS tenant needs to host on UK soil. This is driven by regulatory compliance and the hosting needs of selected existing GOV.UK PaaS tenants who are making preparations for Brexit.
+
+Longer term we will migrate tenants to the London region and demise the region in Ireland to further reduce our footprint.
+
+In addition we need to responsibly manage our use of Amazon Web Service resources to reduce the ongoing running costs. Having duplicate staging environments doubles the amount of infrastructure and hence cost.
```
Comment (samcrang) [inline source/architecture_decision_records/ADR030-single-staging-london.html.md]: Can we just say "Running two staging environments approximately doubles our costs"?

File: source/architecture_decision_records/ADR030-single-staging-london.html.md [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+
+The London region is newer than the Ireland and offers a slightly reduced subset of services compared to the more mature Ireland region.
+
+We feel the risks of running a single staging environment are offset by the cost savings and the reduction in footprint.
```
Comment (samcrang) [inline source/architecture_decision_records/ADR030-single-staging-london.html.md]: I don't feel like we discuss the risks at all before coming to this conclusion.

File: source/architecture_decision_records/ADR030-single-staging-london.html.md (hunk 6 above)
Comment (pauldougan) [reply source/architecture_decision_records/ADR030-single-staging-london.html.md]: both good points

File: source/architecture_decision_records/ADR030-single-staging-london.html.md (hunk 5 above)
Comment (pauldougan) [reply source/architecture_decision_records/ADR030-single-staging-london.html.md]: i have removed this

**Pull request merged**